lychee-lib/src/quirks/mod.rs

```diff
@@ -86,6 +90,12 @@ impl Quirks {
     }
 }
 
+impl Chainable<Request, Status> for Quirks {
+    fn handle(&mut self, input: Request) -> ChainResult<Request, Status> {
```

Comment:
Random thought: should we call it `chain` instead of `handle`? Might align more with the trait name, but I could understand if we don't want to go there.